compositor_render/src/transformations/builtin/node.rs

```diff
@@ -0,0 +1,183 @@
+use std::{sync::Arc, time::Duration};
+
+use compositor_common::scene::{NodeId, Resolution};
+use wgpu::util::DeviceExt;
+
+use crate::{renderer::texture::NodeTexture, transformations::shader::Shader};
+
+use super::{params::BuiltinParams, Builtin, InputState};
+
+struct ConstructedBuiltinNode {
+    shader: Arc<Shader>,
+    builtin: Builtin,
+}
+
+struct ConfiguredBuiltinNode {
+    shader: Arc<Shader>,
+    builtin: Builtin,
+    params_bind_group: wgpu::BindGroup,
+    params_buffer: wgpu::Buffer,
+    input_resolutions: Vec<Option<Resolution>>,
+    output_resolution: Resolution,
+    clear_color: Option<wgpu::Color>,
+}
+
+impl ConfiguredBuiltinNode {
+    fn new(
+        constructed: &ConstructedBuiltinNode,
+        input_resolutions: Vec<Option<Resolution>>,
+    ) -> Self {
+        let output_resolution = constructed.builtin.output_resolution(&input_resolutions);
+        let params = BuiltinParams::new(&constructed.builtin, &input_resolutions);
+        let wgpu_ctx = constructed.shader.wgpu_ctx.clone();
+
+        // This could be created on ConstructedBuiltinNode initialization, but we would need
```

Comment:
You are inside ConstructedBuiltinNode constructor, so this is initialization. Can you clarify the comment?